walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -20,85 +20,125 @@
 import java.util.Collections;
 
 public final class FindMeetingQuery {
+  
+  // Gets the available times for both the required and optional attendees, then returns the overlapping times
+  // which consider optional attendees.
+  public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
+    long duration = request.getDuration();
+
+    Collection<String> requiredAttendees = request.getAttendees();
+    Collection<TimeRange> availableTimes = findFreeTimes(events, requiredAttendees, duration);
+
+    Collection<String> optionalAttendees = request.getOptionalAttendees();
+    Collection<TimeRange> availableTimesOptional= findFreeTimes(events, optionalAttendees, duration);
+
+    Collection<TimeRange> result = findOverlappingTimes(requiredAttendees, availableTimes, availableTimesOptional);
 
-  // Algorithm Complexity: 
-  // Worst case: O(n^2)
+    return result;
+  }
+  
   // Solution: a greedy algorithm which optimizes largest meeting times. Approaches the problem
   // locally, choosing the earliest start time and latest end time, given the next meeting.
-  public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
+  public Collection<TimeRange> findFreeTimes(Collection<Event> events, Collection<String> attendees, long duration) {
```

Comment:
You don't need to change anything but in general, you want to use the tightest type. Instead of `Collection`, you can `List`.